version: '3.8'

services:
  postgres_reg:
    image: postgres:latest
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_DB=users
      - POSTGRES_USER=admin
      - POSTGRES_PASSWORD=admin123
    volumes:
      - postgres_reg_data:/var/lib/postgresql/data
    networks:
      - app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U admin"]
      interval: 10s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4:latest
    ports:
      - "5050:80"
    environment:
      - PGADMIN_DEFAULT_EMAIL=admin@admin.com
      - PGADMIN_DEFAULT_PASSWORD=admin123
    volumes:
      - pgadmin_reg_data:/var/lib/pgadmin
    networks:
      - app-network
    depends_on:
      - postgres_reg

  backend:
    build:
      context: ./backend
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
    environment:
      - DB_HOST=value
      - DB_USER=value
      - DB_PASS=value
    depends_on:
      - postgres_reg
    networks:
      - app-network

  frontend:
    build:
      context: ./frontend
      dockerfile: Dockerfile
    ports:
      - "80:80"
    # depends_on:
    #   - backend
    networks:
      - app-network
      


networks:
  app-network:
    driver: bridge

volumes:
  postgres_reg_data:
  pgadmin_reg_data: